src/RemoteRequest/TemporaryFileCachedRemoteGetRequest.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+namespace AmpProject\RemoteRequest;
+
+use AmpProject\Exception\FailedToGetCachedResponse;
+use AmpProject\RemoteGetRequest;
+use AmpProject\Response;
+use DateTimeImmutable;
+use Exception;
+
+/**
+ * Temporarily cache remote response.
+ *
+ * @package ampproject/amp-toolbox
+ */
+final class TemporaryFileCachedRemoteGetRequest implements RemoteGetRequest
+{
+
+    /**
+     * Prefix to use to identify cached file.
+     *
+     * @var string
+     */
+    const CACHED_FILE_PREFIX = 'amp-remote-request-';
+
+    /**
+     * Cache control header directive name.
+     *
+     * @var string
+     */
+    const CACHE_CONTROL = 'Cache-Control';
+
+    /**
+     * Default cache expiration time in seconds.
+     *
+     * The number represents one month time in seconds. It will be used by default for successful requests when
+     * the 'cache-control: max-age' was not provided.
+     *
+     * @var int
+     */
+    const EXPIRY_TIME = 2592000;
+
+    /**
+     * The decorated RemoteGetRequest instance.
+     *
+     * @var RemoteGetRequest
+     */
+    private $remoteGetRequest;
+
+    /**
+     * Absolute path to the directory where temporary files are saved.
+     *
+     * @var string
+     */
+    private $directory;
+
+    /**
+     * Instantiate a TemporaryFileCachedRemoteGetRequest object.
+     *
+     * @param RemoteGetRequest $remoteGetRequest The decorated RemoteGetRequest object.
+     * @param string           $directory        Optional. Absolute path to the directory where temporary files are
+     *                                           saved.
+     */
+    public function __construct(RemoteGetRequest $remoteGetRequest, $directory = '')
+    {
+        $this->remoteGetRequest = $remoteGetRequest;
+        $this->directory        = $directory ? $directory : sys_get_temp_dir();
+    }
+
+    /**
+     * Get the cached request from a temporary file.
+     *
+     * @param string $url     URL to get.
+     * @param array  $headers Optional. Associative array of headers to send with the request. Defaults to empty array.
+     * @return Response Response for the executed request.
+     * @throws FailedToGetCachedResponse If retrieving the contents from the cache failed.
+     */
+    public function get($url, $headers = [])
+    {
+        $file = $this->getTemporaryFilePath($url);
+
+        if (! file_exists($file)) {
+            return $this->getRemoteResponse($url, $headers);
+        }
+
+        $cachedResponse = file_get_contents($file);
+
+        if (false !== $cachedResponse) {
```

Comment:
We can avoid Yoda conditions on this project:
```suggestion
        if ($cachedResponse !== false) {
```